src/retasc/product_pages_api.py

```diff
@@ -0,0 +1,146 @@
+        https://{pp_release_url}{release_short_name}/schedule-tasks?fields=a,b,c
+        The response have the following fields:name,,date_start,date_finish,flags
+        """
+        res = _no_auth_session.get(
+            f"{self.pp_release_url}{release_short_name}/schedule-tasks",
+            params={
+                "fields": fields,
+            },
+        )
+        return res.json()
+
+    @lru_cache(maxsize=None)
+    @staticmethod
+    def release_schedules_by_flags(
+        self,
+        release_short_name: str,
+        flag_cond: str = "flags_or__in",
+        flags: str = "rcm,releng,exd,sp",
+        fields: str = "release_shortname,name,date_start,date_finish",
+    ) -> list[dict]:
+        """
+        https://{pp_release_url}{release_short_name}/schedule-tasks/?{flag_cond}={flags}&fields=a,b,c
+        Only one flag in condition use `flags_in=sp`
+        Multiple flags in OR condition use `flags_or__in=sp,rcm,releng`
+        Multiple flags in AND condition use `flags_and__in=sp,ga`
+        The response have the following fields:release_shortname,name,slug,date_start,date_finish
+        """
+        res = _no_auth_session.get(
+            f"{self.pp_release_url}{release_short_name}/schedule-tasks",
+            params={
+                flag_cond: flags,
+                "fields": fields,
+            },
+        )
+        return res.json()
+
+    @lru_cache(maxsize=None)
+    @staticmethod
+    def release_schedules_by_name_search(
+        self,
+        release_short_name: str,
+        condition: str,
+        fields: str = "release_shortname,name,slug,date_start,date_finish",
+    ) -> list[dict]:
+        """
+        https://{pp_release_url}{release_short_name}/schedule-tasks?name__regex={condition}&fields=a,b,c
+        The response have the following fields:release_shortname,name,slug,date_start,date_finish
+        """
+        res = _no_auth_session.get(
+            f"{self.pp_release_url}{release_short_name}/schedule-tasks",
+            params={
+                "name__regex": condition,
+                "fields": fields,
+            },
+        )
+        return res.json()
+
+    @lru_cache(maxsize=None)
+    @staticmethod
+    def get_all_active_releases_tasks_for_product(self, product_shortname: str) -> dict:
+        """
+        Get all release names that are still supported by rest api:
+        https://{pp_release_url}?product__shortname={product_shortname}&phase__lt=1000&fields=shortname
+        Loop through the releases and get the tasks of it by rest api
+        https://{pp_release_url}{release_short_name}/schedule-tasks/fields=name,slug,date_start,date_finish
+        """
+
+        product_shortname = product_shortname.strip()
+        if product_shortname not in constants.SUPPORTED_PRODUCTS:
+            raise ValueError(
+                f"Product - {product_shortname} is not supported yet. The products we supported currently are {constants.SUPPORTED_PRODUCTS}"
+            )
```

Comment:
Why is this needed? Can we just support any product?

Reply:
Not all, we only supported those(https://spaces.redhat.com/pages/viewpage.action?spaceKey=EXDSPRHEL&title=SP+RHEL+RelOps+Upcoming+work) and Some of the product will be sunset as well and can be removed later.
Only support what I have tested, not others, can be extend in future?

Reply:
But why limit the supported products here? We can remove/disallow any unsupported products in our internal configuration, right?

Reply:
sure, let me remove this